packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -367,276 +471,278 @@ private void createCaptureSessionWithSessionConfig(
-                setPreCaptureStartTime();
-              }
-              break;
-            case waitingPreCaptureReady:
-              if (aeState == null || aeState != CaptureRequest.CONTROL_AE_STATE_PRECAPTURE) {
-                runPictureCapture();
-              } else {
-                if (hitPreCaptureTimeout()) {
-                  unlockAutoFocus();
-                }
-              }
-          }
-        }
-      };
+      previewRequestBuilder.set(
+          CaptureRequest.CONTROL_AE_PRECAPTURE_TRIGGER,
+          CaptureRequest.CONTROL_AE_PRECAPTURE_TRIGGER_START);
 
-  private void runPictureAutoFocus() {
-    assert (pictureCaptureRequest != null);
+      // Trigger one capture to start AE sequence
+      captureSession.capture(
+          previewRequestBuilder.build(), cameraCaptureCallback, backgroundHandler);
 
-    pictureCaptureRequest.setState(PictureCaptureRequest.State.focusing);
-    lockAutoFocus(pictureCaptureCallback);
+    } catch (CameraAccessException e) {
+      e.printStackTrace();
+    }
   }
 
-  private void runPicturePreCapture() {
-    assert (pictureCaptureRequest != null);
-    pictureCaptureRequest.setState(PictureCaptureRequest.State.preCapture);
-
-    captureRequestBuilder.set(
-        CaptureRequest.CONTROL_AE_PRECAPTURE_TRIGGER,
-        CaptureRequest.CONTROL_AE_PRECAPTURE_TRIGGER_START);
-
-    refreshPreviewCaptureSession(
-        () ->
-            captureRequestBuilder.set(
-                CaptureRequest.CONTROL_AE_PRECAPTURE_TRIGGER,
-                CaptureRequest.CONTROL_AE_PRECAPTURE_TRIGGER_IDLE),
-        (code, message) -> pictureCaptureRequest.error(code, message, null));
-  }
+  /**
+   * Capture a still picture. This method should be called when we get a response in {@link
+   * #cameraCaptureCallback} from both lockFocus().
+   */
+  private void takePictureAfterPrecapture() {
+    Log.i(TAG, "captureStillPicture");
+    cameraCaptureCallback.setCameraState(CameraState.STATE_CAPTURING);
 
-  private void runPictureCapture() {
-    assert (pictureCaptureRequest != null);
-    pictureCaptureRequest.setState(PictureCaptureRequest.State.capturing);
     try {
-      final CaptureRequest.Builder captureBuilder =
+      if (null == cameraDevice) {
+        return;
+      }
+      // This is the CaptureRequest.Builder that we use to take a picture.
+      final CaptureRequest.Builder stillBuilder =
           cameraDevice.createCaptureRequest(CameraDevice.TEMPLATE_STILL_CAPTURE);
-      captureBuilder.addTarget(pictureImageReader.getSurface());
-      captureBuilder.set(
+      stillBuilder.addTarget(pictureImageReader.getSurface());
+
+      // Zoom
+      stillBuilder.set(
           CaptureRequest.SCALER_CROP_REGION,
-          captureRequestBuilder.get(CaptureRequest.SCALER_CROP_REGION));
-      captureBuilder.set(
+          previewRequestBuilder.get(CaptureRequest.SCALER_CROP_REGION));
+
+      // Update builder settings
```

Comment:
This comment isn't very useful since a) it's the name of the method call and b) it tells me nothing about how this differs from the code just above and just below, both of which are *also* updating the builder.